coordinator/core/core.go

```diff
@@ -525,24 +535,36 @@ func (c *Core) generateCertificateForSecret(secret manifest.Secret, parentCertif
 }
 
 func (c *Core) setCAData(dnsNames []string) error {
-	rootCert, rootPrivK, err := generateCert(dnsNames, coordinatorName, nil, nil)
+	rootCert, rootPrivK, err := generateCert(dnsNames, coordinatorName, nil, nil, nil)
 	if err != nil {
 		return err
 	}
-	intermediateCert, intermediatePrivK, err := generateCert(dnsNames, coordinatorIntermediateName, rootCert, rootPrivK)
+	// Creating a cross-signed intermediate cert. See https://github.com/edgelesssys/marblerun/issues/175
+	intermediateCert, intermediatePrivK, err := generateCert(dnsNames, coordinatorIntermediateName, nil, rootCert, rootPrivK)
 	if err != nil {
 		return err
 	}
-	if err := c.store.putCertificate("root", rootCert); err != nil {
+	marbleRootCert, _, err := generateCert(dnsNames, coordinatorIntermediateName, intermediatePrivK, nil, nil)
+	if err != nil {
+		return err
+	}
+
+	if err := c.store.putCertificate(rootCAStore, rootCert); err != nil {
+		return err
+	}
+	if err := c.store.putCertificate(intermediateCAStore, intermediateCert); err != nil {
+		return err
+	}
+	if err := c.store.putCertificate(marbleRootCAStore, marbleRootCert); err != nil {
 		return err
 	}
-	if err := c.store.putCertificate("intermediate", intermediateCert); err != nil {
+	if err := c.store.putPrivK(rootCAStore, rootPrivK); err != nil {
 		return err
 	}
-	if err := c.store.putPrivK("root", rootPrivK); err != nil {
+	if err := c.store.putPrivK(intermediateCAStore, intermediatePrivK); err != nil {
 		return err
 	}
-	if err := c.store.putPrivK("intermediate", intermediatePrivK); err != nil {
+	if err := c.store.putPrivK(marbleRootCAStore, intermediatePrivK); err != nil {
 		return err
 	}
```

Comment:
```suggestion
```
I think we should store this only once (single source of truth). Maybe under a name that is neither "intermediate" nor "marbleRoot".